src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> previous;
+        private Node<T> next;
+
+        public Node(Node<T> previous, T element, Node<T> next) {
+            this.previous = previous;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size > 0) {
+            tail.next = newNode;
+        } else {
+            head = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("The index is wrong");
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
```

Comment:
if (index == size) - does it look familiar? like `add(T value)` method? also, all this can be simplified. you can find node on index and node before it. and all you need to do check whenever previous node is null or not

Reply:
what about this point? **also, all this can be simplified. you can find node on index and node before it. and all you need to do check whenever previous node is null or not**

Reply:
Hope, I understood you right! Please check last commit.